src/super_gradients/training/models/segmentation_models/segformer.py

```diff
@@ -0,0 +1,516 @@
+            == len(eff_self_att_heads)
+            == len(overlap_patch_size)
+            == len(overlap_patch_stride)
+            == len(overlap_patch_pad)
+        ):
+            raise ValueError("All backbone hyper-parameters should be lists of the same length")
+
+        # Patch embeddings
+        self.patch_embed = []
+        for stage_num in range(len(embed_dims)):
+            self.patch_embed.append(
+                PatchEmbedding(
+                    in_channels=in_channels if stage_num == 0 else embed_dims[stage_num - 1],
+                    out_channels=embed_dims[stage_num],
+                    patch_size=overlap_patch_size[stage_num],
+                    stride=overlap_patch_stride[stage_num],
+                    padding=overlap_patch_pad[stage_num],
+                )
+            )
+            self.add_module(f"patch_embed{stage_num+1}", self.patch_embed[stage_num])
+
+        drop_path_rate = 0.1
+        dpr = [x.item() for x in torch.linspace(0, drop_path_rate, sum(encoder_layers))]
+
+        self.blocks = []
+        self.norms = []
+
+        layer_idx = 0
+        for stage_num in range(len(embed_dims)):
+            self.blocks.append(
+                nn.ModuleList(
+                    [
+                        EncoderBlock(
+                            dim=embed_dims[stage_num],
+                            head=eff_self_att_heads[stage_num],
+                            sr_ratio=eff_self_att_reduction_ratio[stage_num],
+                            dpr=dpr[layer_idx + i],
+                        )
+                        for i in range(encoder_layers[stage_num])
+                    ]
+                )
+            )
+            self.norms.append(nn.LayerNorm(embed_dims[stage_num]))
+
+            self.add_module(f"block{stage_num + 1}", self.blocks[stage_num])
+            self.add_module(f"norm{stage_num + 1}", self.norms[stage_num])
+
+            layer_idx += encoder_layers[stage_num]
+
+    def forward(self, x: torch.Tensor) -> List[torch.Tensor]:
+        b_size = x.shape[0]
+
+        features = []
+        for stage_num in range(len(self.patch_embed)):
+            x, h, w = self.patch_embed[stage_num](x)
+
+            for enc_block in self.blocks[stage_num]:
+                x = enc_block(x, h, w)
+            x = self.norms[stage_num](x)
+            x = x.reshape(b_size, h, w, -1).permute(0, 3, 1, 2)
+
+            features.append(x)
+
+        return features
+
+
+class MLP(nn.Module):
+    def __init__(self, dim, embed_dim):
+        """
+        A single Linear layer, with shape pre-processing
+        :param dim: input dimension
+        :param embed_dim: output dimension
+        """
+
+        super().__init__()
+
+        self.proj = nn.Linear(dim, embed_dim)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = x.flatten(2).transpose(1, 2)
+        x = self.proj(x)
+
+        return x
+
+
+class LinearFuse(nn.Module):
```

Comment:
please use the `super_gradients.modules.conv_bn_relu_block.ConvBNReLU` block instead of this

Reply:
done